### PR TITLE
The following PR changes how the master kubeconfig is loaded

### PR DESCRIPTION
**Pull request by timothysc**
and eliminates the kubeconfig settings in sonobuoys config struct.

I still need to finish my testing on it and am breaking up commits to each part. 

/cc @liztio

### CODE REVIEW

File: cmd/sonobuoy/app/kubeconfig.go [hunk 1]
```diff
@@ -37,6 +37,14 @@ func (c *Kubeconfig) Set(str string) error {
 
 // Get returns a rest Config, possibly based on a provided config
 func (c *Kubeconfig) Get() (*rest.Config, error) {
+
+	// detect if in-cluster configuration makes sense
+	if len(c.ExplicitPath) > 0 {
```
Comment (timothysc) [inline cmd/sonobuoy/app/kubeconfig.go]: whoops this is backwards

File: cmd/sonobuoy/app/master.go [hunk 2]
```diff
@@ -25,9 +25,12 @@ import (
 	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
 	"github.com/spf13/cobra"
+
```
Comment (liztio) [inline cmd/sonobuoy/app/master.go]: why the space here?

File: cmd/sonobuoy/app/kubeconfig.go (hunk 1 above)
Comment (liztio) [reply cmd/sonobuoy/app/kubeconfig.go]: probably easier just to compare to empty string.

File: cmd/sonobuoy/app/master.go (hunk 2 above)
Comment (timothysc) [reply cmd/sonobuoy/app/master.go]: logical separator from the line below.

File: cmd/sonobuoy/app/kubeconfig.go [hunk 3]
```diff
@@ -37,6 +37,14 @@ func (c *Kubeconfig) Set(str string) error {
 
 // Get returns a rest Config, possibly based on a provided config
 func (c *Kubeconfig) Get() (*rest.Config, error) {
+
+	// detect if in-cluster configuration makes sense
```
Comment (chuckha) [inline cmd/sonobuoy/app/kubeconfig.go]: This is already handled by https://github.com/heptio/sonobuoy/blob/master/cmd/sonobuoy/app/kubeconfig.go#L46

That call to ClientConfig will return in cluster config if it needs to.

**Pull request merged**